tests/library/har.spec.ts

```diff
@@ -828,3 +828,37 @@ it('should not hang on resources served from cache', async ({ contextFactory, se
   else
     expect(entries.length).toBe(2);
 });
+
+it('should not hang on slow chunked response', async ({ browserName, browser, contextFactory, server }, testInfo) => {
+  it.info().annotations.push({ type: 'issue', description: 'https://github.com/microsoft/playwright/issues/21182' });
+  it.fixme(browserName === 'webkit');
+  server.setRoute('/empty.html', (req, res) => {
+    res.writeHead(200, {
+      'Content-Type': 'text/html',
+    });
+    res.end(`<script>
+    let receivedFirstData = new Promise(f => {
```

Comment:
we are not actually waiting for the progress event correctly when waiting on this promise, since `receivedFirstData` would be undefined. needs to be changed from `let ->  var`